lib/Log.js

```diff
@@ -1,155 +1,231 @@
 'use strict';
 
-const raven = require('raven');
+const HomeyModule = require('homey');
 
-class Log {
-
-	constructor() {
-
-		this._capturedExceptions = [];
-		this._capturedMessages = [];
-
-		if( typeof global.Homey === 'undefined' ) {
-			try {
-				this._homey = require('homey');
-			} catch( err ) {
-				console.error(err)
-				return console.error('Error: Homey not found');
-			}
-		} else {
-			this._homey = global.Homey;
-		}
-		
-		if( typeof this._homey === 'undefined' )
-			return console.error('Error: Homey not found');
-
-		if( typeof this._homey.env.HOMEY_LOG_URL === 'string' ) {
-			this.init( this._homey.env.HOMEY_LOG_URL );
-		}
-
-	}
-
-	_log() {
-		console.log.bind( null, Log.logTime(), '[homey-log]' ).apply( null, arguments );
-	}
-
-	init( url, opts ) {
-
-		if( process.env.DEBUG === '1' && this._homey.env.HOMEY_LOG_FORCE !== '1' )
-			return this._log('App is running in debug mode, disabling log');
-
-		this._client = new raven.Client( url, opts );
-
-		this._client.patchGlobal();
-		
-		this.setTags({
-			appId			: this._homey.manifest.id,
-			appVersion		: this._homey.manifest.version,
-			homeyVersion	: this._homey.version
-		});
-
-		if (this._homey.hasOwnProperty('ManagerCloud')) { // SDKv2
-			this._homey.ManagerCloud.getHomeyId(this.setHomeyIdTag.bind(this))
-		} else if (this._homey.hasOwnProperty('manager')) { // SDKv1
-			this._homey.manager('cloud').getHomeyId(this.setHomeyIdTag.bind(this))
-		}
-
-		this._log(`App ${this._homey.manifest.id} v${this._homey.manifest.version} logging...`);
-
-		return this;
-
-	}
-
-	setHomeyIdTag(err, result) {
-		if (!err && typeof result === 'string') {
-			this.setTags({ homeyId: result })
-		}
-	}
-
-	setTags( tags ) {
-		if( this._client ) {
-			this._client.setTagsContext(tags);
-		}
+const Raven = require('raven');
```

Comment:
not relevant for this PR but should we upgrade to `@sentry/node` ?

Reply:
Yes, good question, that was also suggested here https://github.com/athombv/node-homey-log/pull/6, however, it appears to have an install size of ~4.2MB which I think is too much for a Homey App. Maybe if they manage to get the install size down to a reasonable number we can consider to upgrade.

Reply:
Yea its install size is HUGE but its actual bundle size is apparently not that big...
 https://packagephobia.com/result?p=%40sentry%2Fnode
Seems like it is shipping a bunch of extra files and their monorepo structure doesn't help in this case :/

So we could use the latest and greatest but only if we use a bundler